source/test/F0.Minesweeper.Components.Tests/Logic/Game/GameUpdaterTests.cs

```diff
@@ -0,0 +1,116 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Bunit;
+using F0.Minesweeper.Components.Abstractions;
+using F0.Minesweeper.Logic.Abstractions;
+using FluentAssertions;
+using Microsoft.Extensions.DependencyInjection;
+using Moq;
+using Xunit;
+
+using GameCell = F0.Minesweeper.Components.Pages.Game.Modules.Cell;
+
+namespace F0.Minesweeper.Components.Logic.Game
+{
+	public class GameUpdaterTests
+	{
+		[Fact]
+		public void WithReport_SetsReport()
+		{
+			// Arrange
+			Mock<IGameUpdateReport> gameReport = new(MockBehavior.Strict);
+			gameReport.Setup(report => report.Cells).Throws<NotImplementedException>();
+			GameUpdaterForTests instanceUnderTest = new();
+
+			// Act
+			instanceUnderTest.WithReport(gameReport.Object);
+
+			// Assert
+			Action actionUnderAssertion = () => { var cells = instanceUnderTest.ExposedReport?.Cells; };
+			actionUnderAssertion.Should().Throw<NotImplementedException>();
+		}
+
+		[Fact]
+		public void UpdateAsync_NoReportSet_Throws()
+		{
+			// Arrange
+			GameUpdaterForTests instanceUnderTest = new();
+
+			// Act & Assert
+			Func<Task> actionUnderTest = async () => await instanceUnderTest.UpdateAsync(Enumerable.Empty<Pages.Game.Modules.Cell>(), new Location(1, 1));
+			actionUnderTest.Should().Throw<InvalidOperationException>();
+		}
+
+        [Fact]
+		public async Task UpdateAsync_WithReport_CallsOnUpdateWithUncoverableCells()
+		{
+			// Arrange
+
+			// Game Field 	=> x = 5, y = 3
+			// Clicked 		=> x = 2, y = 1
```

Comment:
Great testing!
I can only comment on spacing and I put my comment here:
There seems to be mixed tabs and spaces in some test-files, which look weird on _GitHub_

Reply:
Went through all the test files in the components again and formatted them